cmd/mailer.go

```diff
@@ -51,11 +42,22 @@ func runSendMail(c *cli.Context) error {
 		}
 	}
 
-	mailer.NewContext()
-	msg := mailer.NewMessage(emails, subject, body)
-	err = mailer.SendSync(msg)
+	var emails []string
+	err := models.IterateUser(func(user *models.User) error {
+		emails = append(emails, user.Email)
+		return nil
+	})
 	if err != nil {
-		return err
+		return errors.New("Cann't find users")
+	}
+
+	mailer.NewContext()
+	for _, email := range emails {
+		msg := mailer.NewMessage([]string{email}, subject, body)
+		err = mailer.SendSync(msg)
+		if err != nil {
+			return err
+		}
```

Comment:
I think sending the messages 1 by 1 will be time-consuming especially for instances with huge amount of users.

We can put these in goroutines instead?

Reply:
f5a108d

Reply:
By the way I am not sure about this commit.

It may be not save to do it in parallel? Otherwise why SendAsync uses a queue instead of just spawning a goorutine with send

https://github.com/go-gitea/gitea/blob/b222dbc1d1df2559206ec1c27928d4666a48f0e8/services/mailer/mailer.go#L338-L352

Reply:
I think `SendAsync` will be better then. This usecase is already covered. Better than rewriting out own functionality

Reply:
Nice stuff :)

Reply:
But there's no way to wait before each one is sent in such case.

Reply:
Since we have workers go routines for queue, the go routine is unnecessary. Just change it to 

```
func SendAsync(msg *Message) { 
 	_ = mailQueue.Push(msg) 
 } 
```

will be OK.

Reply:
Agreed. Good catch by the way.

But @lunny am I wrong that if we will push it in the queue and terminate the app most like a bunch of the messages will be missed?

If you missed the context I am trying to send emails from from cmd command. And as I understand if it will use this queue there's no way to way before it's finished.

Reply:
@zhiburt Yeah. I'm out of the topic. I think we should do that via internal API but not push to the queue directly as @zeripath said.